### PR TITLE
Draft code of modbus mapper

### PR DESCRIPTION
**Pull request by sailorvii**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
> /kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: mappers/common/configmaptype.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+// PropertyVisitor is structure to store propertyVisitor in deviceProfile.json in configmap
+type PropertyVisitor struct {
+	Name          string `json:"name,omitempty"`
+	PropertyName  string `json:"propertyName,omitempty"`
+	ModelName     string `json:"modelName,omitempty"`
+	CollectCycle  int64  `json:"collectCycle"`
+	ReportCycle   int64  `json:"reportcycle,omitempty"`
+	PProperty     Property
```
Comment (luogangyi) [inline mappers/common/configmaptype.go:79]: what is PProperty used for? and why there is no json annotations？

File: mappers/common/data_converter.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import (
+	"errors"
+	"strconv"
+)
+
+// String to other types
+func Convert(t string, value string) (r interface{}, err error) {
+	switch t {
+	case "int":
+		return strconv.ParseInt(value, 10, 64)
```
Comment (luogangyi) [inline mappers/common/data_converter.go:28]: return strconv.ParseInt(value, 10, 64), nil

File: mappers/common/event.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+	if tc := mc.Client.Connect(); tc.Wait() && tc.Error() != nil {
+		return tc.Error()
+	}
+
+	mc.Qos = 0          // At most 1 time
+	mc.Retained = false // Not retained
+	return nil
+}
+
+func (mc *MqttClient) GetConnection() error {
```
Comment (luogangyi) [inline mappers/common/event.go]: seems duplicated with Connect() and no place to use it?

File: mappers/common/event.go [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+}
+
+func CreateMessageTwinUpdate(name string, valueType string, value string) []byte {
+	var updateMsg DeviceTwinUpdate
+	updateMsg.BaseMessage.Timestamp = getTimestamp()
+	updateMsg.Twin = map[string]*MsgTwin{}
+	updateMsg.Twin[name] = &MsgTwin{}
+	updateMsg.Twin[name].Actual = &TwinValue{Value: &value}
+	updateMsg.Twin[name].Metadata = &TypeMetadata{Type: valueType}
+
```
Comment (luogangyi) [inline mappers/common/event.go:98]: why not return Error?

File: mappers/common/event.go [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+	msg, err := json.Marshal(dataMsg)
+	if err != nil {
+		return make([]byte, 0)
+	}
+	return msg
+}
+
+func CreateMessageState(state string) []byte {
+	var stateMsg DeviceUpdate
```
Comment (luogangyi) [inline mappers/common/event.go:120]: BaseMessage does not need initialization?

File: mappers/common/timer.go [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+	Times    int
+}
+
+func (t *Timer) Start() {
+	ticker := time.NewTicker(t.Duration)
+	if t.Times > 0 {
+		for i := 0; i < t.Times; i++ {
+			select {
+			case <-ticker.C:
+				t.Function()
```
Comment (luogangyi) [inline mappers/common/timer.go:37]: go t.Function()

maybe better

File: mappers/modbus/configmap/parse.go [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+	"errors"
+	"io/ioutil"
+
+	mappercommon "github.com/kubeedge/kubeedge/mappers/common"
+	"github.com/kubeedge/kubeedge/mappers/modbus/globals"
+	"k8s.io/klog"
+)
+
+func Parse(path string,
+	devices map[string]*globals.ModbusDev,
```
Comment (luogangyi) [inline mappers/modbus/configmap/parse.go:31]: define them as global var or return value?
seems no way to use these var now

**Review by luogangyi — COMMENTED** [review]
part of review

File: mappers/common/configmaptype.go (hunk 1 above)
Comment (sailorvii) [reply mappers/common/configmaptype.go:79]: this visitor is in the device structure. The device could get the property directly.

File: mappers/common/data_converter.go (hunk 2 above)
Comment (sailorvii) [reply mappers/common/data_converter.go:28]: Done

File: mappers/common/event.go (hunk 3 above)
Comment (sailorvii) [reply mappers/common/event.go]: Yes, remove it

File: mappers/common/event.go (hunk 4 above)
Comment (sailorvii) [reply mappers/common/event.go:98]: Refined.

File: mappers/common/event.go (hunk 5 above)
Comment (sailorvii) [reply mappers/common/event.go:120]: Added

**Pull request merged**